woocommerce/woocommerce-gutenberg-products-block
# Extensibility for Block Checkout Address Fields

This is an extensibility enhancement for the Checkout block which fixes #3633.

When developing Checkout we decided to include a list of country locales in Blocks to support an improved format, and allow us to fully control the field locales. This however meant that plugins/users extending core address fields, or adding custom locales (including country states/regions), were not honoured.

This PR proposes a change which imports the locales from core, changes them to our new format, and merges them to create the correct address field config. In order of priority it merges Default Fields > Locale Fields > Overrides at block level.

Fixes #3633

### How to test the changes in this Pull Request:

1. Change the country on the checkout and check fields are updated/sorted. Japan is a good example because it re-orders all fields. US has states. UK has a text input for state.
2. Install https://wordpress.org/plugins/portugal-states-distritos-for-woocommerce/ Confirm states appear for portugal
3. Complete checkout - no errors, correct address stored.

### Changelog

> Dev - Block Checkout will now respect custom address locales and custom country states via core filter hooks.
